frame/democracy/src/lib.rs

```diff
@@ -998,7 +998,8 @@ decl_module! {
 			ensure!(now >= since + voting + additional, Error::<T>::TooEarly);
 			ensure!(expiry.map_or(true, |e| now > e), Error::<T>::Imminent);
 
-			let _ = T::Currency::repatriate_reserved(&provider, &who, deposit, BalanceStatus::Free);
+			let res = T::Currency::repatriate_reserved(&provider, &who, deposit, BalanceStatus::Free);
+			debug_assert!(res.is_ok());
```

Comment:
When will `debug_assert!` work? Will it work in production network?

Reply:
An optimized build will not execute debug_assert! statements unless -C debug-assertions is passed to the compiler

Reply:
https://doc.rust-lang.org/std/macro.debug_assert.html